plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java

```diff
@@ -1753,4 +1756,374 @@ private Set<String> getAllFilesFromCdfDirectory(String tableName)
+                ImmutableList.of("('a', '1')", "('b', '2')"))) {
+            assertQuery(session, "SELECT * FROM " + table.getName() + " WHERE CAST (part AS INTEGER) = 1", "VALUES ('a', 1)");
+        }
+    }
+
+    @Test
+    public void testPartitionPredicateInOuterQuery()
+    {
+        Session session = sessionWithPartitionFilterRequirement();
+        try (TestTable table = new TestTable(
+                getQueryRunner()::execute,
+                "test_partition_predicate",
+                "(x INTEGER, part INTEGER) WITH (partitioned_by = ARRAY['part'])",
+                ImmutableList.of("(1, 11)", "(2, 22)"))) {
+            assertQuery(session, "SELECT * FROM (SELECT * FROM " + table.getName() + " WHERE x = 1) WHERE part = 11", "VALUES (1, 11)");
+        }
+    }
+
+    @Test
+    public void testPartitionPredicateInInnerQuery()
+    {
+        Session session = sessionWithPartitionFilterRequirement();
+        try (TestTable table = new TestTable(
+                getQueryRunner()::execute,
+                "test_partition_predicate",
+                "(x INTEGER, part INTEGER) WITH (partitioned_by = ARRAY['part'])",
+                ImmutableList.of("(1, 11)", "(2, 22)"))) {
+            assertQuery(session, "SELECT * FROM (SELECT * FROM " + table.getName() + " WHERE part = 11) WHERE x = 1", "VALUES (1, 11)");
+        }
+    }
+
+    @Test
+    public void testPartitionPredicateAnalyze()
+    {
+        Session session = sessionWithPartitionFilterRequirement();
+        try (TestTable table = new TestTable(
+                getQueryRunner()::execute,
+                "test_partition_predicate_analyze_",
+                "(x INTEGER, part INTEGER) WITH (partitioned_by = ARRAY['part'])",
+                ImmutableList.of("(1, 11)", "(2, 22)"))) {
+            String expectedMessageRegExp = "The session with enabled property delta.query-partition-filter-required does not work with ANALYZE statement.";
+            assertQueryFails(session, "ANALYZE " + table.getName(), expectedMessageRegExp);
+            assertQueryFails(session, "EXPLAIN ANALYZE " + table.getName(), expectedMessageRegExp);
+        }
+
+        try (TestTable nonPartitioned = new TestTable(
+                getQueryRunner()::execute,
+                "test_partition_predicate_analyze_",
+                "(a INTEGER, b INTEGER) ",
+                ImmutableList.of("(1, 11)", "(2, 22)"))) {
+            assertUpdate(session, "ANALYZE " + nonPartitioned.getName());
+            computeActual(session, "EXPLAIN ANALYZE " + nonPartitioned.getName());
+        }
+    }
+
+    @Test
+    public void testPartitionFilterMultiplePartition()
+    {
+        Session session = sessionWithPartitionFilterRequirement();
+        try (TestTable table = new TestTable(
+                getQueryRunner()::execute,
+                "test_partition_filter_multiple_partition_",
+                "(x VARCHAR, part1 INT, part2 INT) WITH (PARTITIONED_BY = ARRAY['part1', 'part2'])",
```

Comment:
Instead of `INT` can we use `INTEGER` 

Based on the commit message - https://github.com/trinodb/trino/pull/18372/commits/a4f5c63e73828357b40aae9069147665ef94a3e7 should we use them as lowercase - a follow-up idea